### PR TITLE
🃏 Model card for TRL

### PR DESCRIPTION
**Pull request by qgallouedec**
# What does this PR do?

Having our own model card.

## Demo

```python
from datasets import load_dataset
from trl import DPOConfig, DPOTrainer
from transformers import AutoModelForCausalLM, AutoTokenizer

model = AutoModelForCausalLM.from_pretrained("Qwen/Qwen2-0.5B-Instruct")
tokenizer = AutoTokenizer.from_pretrained("Qwen/Qwen2-0.5B-Instruct")
dataset_name = "trl-lib/tldr-preference"
train_dataset = load_dataset(dataset_name, split="train").select(range(50))
args = DPOConfig(output_dir="dpo-qwen2", logging_steps=10, push_to_hub=True, save_strategy="epoch", num_train_epochs=1)
trainer = DPOTrainer(model=model, args=args, tokenizer=tokenizer, train_dataset=train_dataset)
trainer.train()

if args.push_to_hub:
    trainer.push_to_hub(dataset_name=dataset_name)
```

result: https://huggingface.co/qgallouedec/dpo-qwen2

### Link to the paper

Screenshot 2024-09-25 at 18 44 17
Screenshot 2024-09-25 at 18 44 22

### Link to the dataset

Screenshot 2024-09-25 at 23 12 53

Screenshot 2024-09-25 at 23 13 04

### TRL own model card
Screenshot 2024-09-26 at 00 18 28

### Other

- Citations
- Links to wandb run

## TODO

- [x] `AlignPropTrainer`
- [x] `BCOTrainer`
- [x] `CPOTrainer`
- [x] `DPOTrainer`
- [x] `GKDTrainer`
- [x] `IterativeSFTTrainer`
- [x] `KTOTrainer`
- [x] `NashMDTrainer`
- [x] `OnlineDPOTrainer`
- [x] `ORPOTrainer`
- [x] `PPOv2Trainer`
- [x] `RewardTrainer`
- [x] `RLOOTrainer`
- [x] `SFTTrainer`
- [x] `XPOTrainer`

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/trl/tree/main/docs).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: examples/scripts/alignprop.py [hunk 1]
```diff
@@ -132,4 +132,4 @@ def image_outputs_logger(image_pair_data, global_step, accelerate_logger):
     # Save and push to hub
     trainer.save_model(training_args.output_dir)
     if training_args.push_to_hub:
-        trainer.push_to_hub()
+        trainer.push_to_hub(dataset_name=script_args.dataset_name)
```
Comment (lewtun) [inline examples/scripts/alignprop.py:135]: Nice, I did not know we could pass this arg in `push_to_hub()`!

File: examples/scripts/rloo/rloo.py [hunk 2]
```diff
@@ -133,6 +133,6 @@ def tokenize(element):
     # Save and push to hub
     trainer.save_model(training_args.output_dir)
     if training_args.push_to_hub:
-        trainer.push_to_hub()
+        trainer.push_to_hub(dataset_name="trl-internal-testing/descriptiveness-sentiment-trl-style")
```
Comment (lewtun) [inline examples/scripts/rloo/rloo.py:136]: Note to self: we should move these datasets that aren't strictly used for tests to `trl-lib`

File: tests/test_dpo_trainer.py [hunk 3]
```diff
@@ -782,7 +782,8 @@ def test_dpo_lora_tags(self):
                 peft_config=lora_config,
             )
 
-            assert trainer.model.model_tags == trainer._tag_names
+            assert "dpo" in trainer.model.model_tags
```
Comment (lewtun) [inline tests/test_dpo_trainer.py:785]: WDYT about gradually moving to using `unittest`'s methods for the tests (like we do in other HF libs)?

In this case, I think you could write:

```python
for tag in ["dpo", "trl"]:
    self.assertIn(tag, trainer.model.model_tags)
```

File: trl/templates/model_card.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+It has been trained using [TRL](https://github.com/huggingface/trl).
+
+## Quick start
+
+```python
+from transformers import pipeline
+
+question = "If you had a time machine, but could only go to the past or the future once and never return, which would you choose and why?"
+generator = pipeline("text-generation", model="{{ hub_model_id }}", device="cuda")
+output = generator([{"role": "user", "content": question}], max_new_tokens=500)[0]
```
Comment (lewtun) [inline trl/templates/model_card.md:17]: 500 tokens might be a bit slow as the default for LLMs, so maybe we can try this:

```suggestion
output = generator([{"role": "user", "content": question}], max_new_tokens=128, return_full_text=False)[0]
```

Note that `return_full_text=False` will just return the completion which I a bit simpler below

File: trl/templates/model_card.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+print(output["generated_text"][1]["content"])
```
Comment (lewtun) [inline trl/templates/model_card.md:18]: If you use the `return_full_text=False` above then you just need this:

```suggestion
print(output["generated_text"])
```

**Review by lewtun — APPROVED** [review]
Amazing work on the model cards @qgallouedec - now they're really packed with useful information 🔥 !!

LGTM with a tweak to the example inference code

File: examples/scripts/alignprop.py (hunk 1 above)
Comment (qgallouedec) [reply examples/scripts/alignprop.py:135]: push_to_hub forwards the kwargs to create_model_card. So if you have this args in create_model_card, you can pass it.

File: trl/templates/model_card.md (hunk 4 above)
Comment (qgallouedec) [reply trl/templates/model_card.md:17]: I think you don't need for pipeline, let me check

**Comment by qgallouedec** [comment]
Note that the card for diffusion models and VLM will be wrong but we can probably keep it like that for now

**Comment by qgallouedec** [comment]
> Note that the code demo for diffusion models and VLM will be wrong but we can probably keep it like that for now

File: trl/templates/model_card.md (hunk 4 above)
Comment (qgallouedec) [reply trl/templates/model_card.md:17]: image

File: tests/test_dpo_trainer.py (hunk 3 above)
Comment (qgallouedec) [reply tests/test_dpo_trainer.py:785]: #2137

Comment (qgallouedec) [reply tests/test_dpo_trainer.py]: [65af404](https://github.com/huggingface/trl/pull/2123/commits/65af404b6ce79f2949596a755007ff8fbbcbfc8d)

**Review by kashif — APPROVED** [review]

**Comment by valayDave** [comment]
When will this commit be released ? Will it be a part of v0.12 ?

**Pull request merged**